src/main/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -1096,4 +1097,79 @@ public static SAMRecord clipOverlappingAlignedBases(final SAMRecord record, fina
+        final Pattern commaPattern = Pattern.compile("[,]");
+        if( record == null ) throw new IllegalArgumentException("record is null");
+        if( record.getHeader() == null ) throw new IllegalArgumentException("record.getHeader() is null");
+        /* extract value of SA tag */
+        final Object saValue = record.getAttribute( SAMTagUtil.getSingleton().SA );
+        if( saValue == null ) return Collections.emptyList();
+        if( ! (saValue instanceof String) ) throw new SAMException(
+                "Expected a String for attribute 'SA' but got " + saValue.getClass() );
+        
+        final SAMRecordFactory samReaderFactory = new DefaultSAMRecordFactory();
+        
+        /* the spec says: "Other canonical alignments in a chimeric alignment, formatted as a 
+         * semicolon-delimited list: (rname,pos,strand,CIGAR,mapQ,NM;)+. 
+         * Each element in the list represents a part of the chimeric alignment.
+         * Conventionally, at a supplementary line, the  1rst element points to the primary line.
+         */
+        
+        /* break string using semicolon */
+        final String semiColonStrs[] = semiColonPattern.split((String)saValue);
+        
+        /* the result list */
+        final List<SAMRecord> alignments = new ArrayList<>( semiColonStrs.length );
+        
+        for( final String semiColonStr : semiColonStrs ) {
+            /* ignore empty string */
+            if( semiColonStr.isEmpty() ) continue;
+            
+            /* break string using comma */
+            final String commaStrs[] = commaPattern.split(semiColonStr);
+            if( commaStrs.length != 6 )  throw new SAMException("Bad 'SA' attribute in " + semiColonStr);
+            
+            /* create the new record */
+            final SAMRecord otherRec = samReaderFactory.createSAMRecord( record.getHeader() );
+            
+            /* copy fields from the original record */
+            otherRec.setReadName( record.getReadName() );
+            otherRec.setReadBases( record.getReadBases() );
+            otherRec.setBaseQualities( record.getBaseQualities() );
+            
+            /* get reference sequence */
+            final int tid = record.getHeader().getSequenceIndex( commaStrs[0] );
+            if( tid == -1 ) throw new SAMException("Unknown contig in " + semiColonStr);
+            otherRec.setReferenceIndex( tid );
+            
+            /* fill other fields */
+            otherRec.setAlignmentStart( Integer.parseInt(commaStrs[1]) ); 
+            otherRec.setFlags(
+                SAMFlag.SUPPLEMENTARY_ALIGNMENT.flag + 
+                (commaStrs[2].equals("+") ? 0 : SAMFlag.READ_REVERSE_STRAND.flag) );
+            otherRec.setCigar( TextCigarCodec.decode( commaStrs[3] ) );
+            otherRec.setMappingQuality( Integer.parseInt(commaStrs[4]) );
+            otherRec.setAttribute( SAMTagUtil.getSingleton().NM , Integer.parseInt(commaStrs[5]) );                  
```

Comment:
All of these .parseInt() calls open the possibility of breaking on bad input and exposing a stack trace to the user. Perhaps you should emit a SAMException if any of these fields don't parse correctly?